### ISSUE TITLE
Invalid TypeScript Type (Default Export)

### ISSUE THREAD

**Issue by ryanelian** [postcss/postcss#1424]
PostCSS exports default type as seen in the type file:

https://github.com/postcss/postcss/blob/master/lib/postcss.d.ts

However that is certainly not the case because a code calling default exported type will not run:

```ts
import postcss from 'postcss';

        const postcssResult = await postcss([autoprefixer])
            .process(sassResult.css, postcssOptions);
```

```
TypeError: postcss_1.default is not a function
```

Instead, shouldn't the TypeScript declaration use `export =` ?

https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require

**Not to mention, calling `import postcss = require('postcss');` from TypeScript will not compile anymore due to this type declaration change!**

----

Workaround for those affected by this issue: revert back to PostCSS 7

**Comment by ai** [comment]
Show your `tsconfig.json`

> TypeError: postcss_1.default is not a function

Am I right that it is an runtime error (and not TS types checking error)? In this case we have a problem with TS compiler, not with PostCSS types.

Maybe we need to fix how TS compile external modules.

**Comment by cekvenich** [comment]
I have same(similar) error. 'This expression is not callable' on something that used to work with the older version on the line:
      postcss([autoprefixer])

My tsconfig is but has not changed, only poscss version changed.
```
{
   "compilerOptions": {
      "target": "ES2019",

      "module": "commonjs",
      "moduleResolution": "node",
      "esModuleInterop" : true,
      "allowSyntheticDefaultImports": true,
      
      "lib": ["ES2018"],

      "strictBindCallApply": true,	
      "noImplicitThis": true,

      "removeComments": false,
      "outDir": ".",
      "baseUrl": ".",
      "paths": {
         "*": ["node_modules/*"]
      }
   },

   "include": [
      "**/*.ts"
   ],
   "exclude": [
      "node_modules",
      "webApp"
   ]
}

```

**Comment by cekvenich** [comment]
I see a new release today. I still have the same error:

vic@vic-OptiPlex-7020:/media/vic/dat/Github2/mbCLI/src$ tsc
lib/Extra.ts:242:7 - error TS2349: This expression is not callable.
  Type 'typeof import("/media/vic/dat/Github2/mbCLI/src/node_modules/postcss/lib/postcss")' has no call signatures.

242       postcss([autoprefixer])
          ~~~~~~~
Found 1 error.

vic@vic-OptiPlex-7020:/media/vic/dat/Github2/mbCLI/src$

**Comment by ai** [comment]
I will need PR for this fix

**Comment by Semigradsky** [comment]
TS doesn't support `.mjs` files yet, so current typings lie.
Typings should be rewritten to commonjs `declare namespace ...` and `export = X` style 😞

**Comment by ai** [comment]
Very strange that we have `default` key in import if TS do not support `.mjs` files.

Can we fix the problem by adding `dual-publish`?

**Comment by Semigradsky** [comment]
Yes, move from "commonjs `.js` and `.mjs`" to "esm `.js` and `.cjs`" should help.
In this case typings will be correct.

**Comment by remcohaszing** [comment]
This is basically the same pitfall as auth0/jwt-decode#103. All possible solutions are mentioned in that issue.

This can’t be properly typed, because type definitions for default exports are not the same as for `module.exports`. The current types are for default exports, which is correct for example for Webpack builds, but not for usage with NodeJS, which uses `module.exports`.

Given the current situation, the only non-breaking solution would be to add the following to each `.js` file:

```js
module.exports.default = module.exports;
```

Or equivalent, but more specifical to each file (this example is based on [postcss.js](https://github.com/postcss/postcss/blob/master/lib/postcss.js)):

```js
postcss.default = postcss;
```

Also `esModuleInterop` needs to be removed from [tsconfig.json](https://github.com/postcss/postcss/blob/master/tsconfig.json). This doesn’t affect `.d.ts` files, but it does give the false idea it does.

**Comment by ai** [comment]
The fix was released in 8.1.7
